### ISSUE TITLE
Have the vault and consul cookbooks been abandoned?

### ISSUE THREAD

**Issue by empath** [sous-chefs/consul#493]
Seems like there's very little activity and they're both several versions behind.

Are there any active forks that have done any work to update to the latest versions?

I'm probably going to fork and start the work myself, but I'd like not to duplicate effort if someone has already done it or is already working on it.

**Comment by gdavison** [comment]
Hi @empath,

I've started (very slowly, admittedly) on adding the new v1.x parameters in a backwards-compatible way

**Comment by johnbellone** [comment]
I have been accepting collaborators and maintainers for the past several months. That includes the management of the releases for the two cookbooks. @gdavison has offered to help out. And we can perform a release if that's what is needed. But I was under the impression that there was some more work that needed to go on before doing it.

**Comment by shayangz** [comment]
A release would help a lot if possible specially for supporting missing config parameters.

**Comment by scalp42** [comment]
The issue is that old parameters were renamed as well (`statsd_addr` vs `statsd_address`). I think we should just rescue missing/invalid params and trust the user, instead of updating cookbook all the time IMO. 

I'd not mind putting a PR together.

In the meantime, I'd port over the default recipe from `consul` cookbook and skip the `consul_config` recipe *or* write whatever needed onfig to `['consul']['service']['config_dir']` and keep the defaults as minimal as possible. The new config in `conf.d` will override whatever defaults.

@shayangz (tweak if using Vault etc):

```ruby
# if you want extra node metadata, feel free to skip otherwise
if node['ec2']
  chef_consul_config = node[cookbook_name]['config'].merge(
    node_meta: {
      instance_type: node['ec2']['instance_type'],
      local_hostname: node['ec2']['local_hostname'],
      local_ipv4: node['ec2']['local_ipv4'],
      instance_id: node['ec2']['instance_id'],
      region: node['ec2']['region'],
      availability_zone: node['ec2']['availability_zone'],
      public_ipv4: node.to_h.dig('ec2', 'public_ipv4'),
      public_hostname: node.to_h.dig('ec2', 'public_hostname'),
      chef_environment: node.chef_environment,
      datacenter: node[cookbook_name]['config']['datacenter'],
    }
  )
else
  chef_consul_config = node[cookbook_name]['config'].merge(
    node_meta: {
      chef_environment: node.chef_environment,
      datacenter: node[cookbook_name]['config']['datacenter'],
    }
  )
end

blacklist = ['acl_master_token']
chef_consul_config = Hash[node[cookbook_name]['config'].reject do |k|
  blacklist.include?(k)
end.sort_by { |key, val| key.to_s }]

# if including consul cookbook, put it in node['consul']['service']['config_dir']
# file ::File.join(node['consul']['service']['config_dir'], 'chef.json') do

file ::File.join('/etc/consul', 'chef.json') do
  owner node['consul']['service_user'] if "grep #{node['consul']['service_user']} /etc/passwd"
  group node['consul']['service_group'] if "grep #{node['consul']['service_group']} /etc/group"
  mode '0640'
  content lazy { JSON.pretty_generate(chef_consul_config , quirks_mode: true) }
  action node[cookbook_name]['nuke'] ? :delete : :create
  notifies :reload, "consul_service[#{node['consul']['service_name']}]", :delayed
end
```

Hopefully it helps!

Remind me of the good old days of ⚙️ 0️⃣ 💪

**Comment by gdavison** [comment]
#496 is work-in-progress

**Comment by scalp42** [comment]
@gdavison thanks for the PR. 

I'd be more leaning toward letting the user handle the actual JSON rather than creating new classes per major (I'm guessing) Consul versions.

Would greatly reduce the churn as well if new attributes are added in minor versions.

Or best of both worlds I guess, simply adding a new `consul_custom_config` resource. What do you think?

**Comment by scalp42** [comment]
I guess that never happened 🌵

**Comment by pierresouchay** [comment]
@scalp42 Completely agree here.

@gdavison @johnbellone Can we help to release new versions of this Cookbook ?

We need those PR in a new release:

 * https://github.com/johnbellone/consul-cookbook/pull/502 discovery_max_stale - merged
 * https://github.com/johnbellone/consul-cookbook/pull/504 Consul Connect support

While the first one has been merged, no new version has been released

**Comment by damacus** [comment]
Hi @scalp42 would you like to help maintain the consul cookbook on a more formal basis, your PRs have been most helpful so far!

**Comment by damacus** [comment]
Closing as I think the question has been answered.

**Comment by pierresouchay** [comment]
@damacus we are not using the community cookbook for the same reasons as @scalp42 but we would be glad to help it improving and handle more easily new features of Consul

**Comment by damacus** [comment]
That would be awesome! 

@pierresouchay please feel free to drop by the sous-chefs room in the Chef community Slack.

In the mean time I think we have a small contingent of people willing to moderise this cookbook.

**Comment by scalp42** [comment]
I'm strongly considering forking the cookbook ala mongodb from back in the days. The new ACL model in Consul feels like a good time to tackle this, removing poise, properly using test-kitchen, etc etc.

I'm not even sure who's leading the development on this cookbook to be transparent as @pierresouchay and I have emitted the same concerns but bad patterns (the whole v0/v1 provider approach) were merged so it's confusing.
